modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -8917,6 +8962,189 @@ public final void fireEvent(Event event) {
+        if (transitionTimers == null) {
+            return null;
+        }
+
+        for (int i = 0, max = transitionTimers.size(); i < max; ++i) {
+            TransitionTimer timer = transitionTimers.get(i);
+
+            // We use an identity comparison here because we're looking for the exact property
+            // instance that was targeted by the transition timer on this node.
+            if (timer.getTargetProperty() == property) {
+                return timer;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Completes all running timers, which skips the rest of their animation and sets
+     * the property to the target value.
+     */
+    // package-private for testing
+    void completeTransitionTimers() {
+        if (transitionTimers == null || transitionTimers.isEmpty()) {
+            return;
+        }
+
+        // Make a copy of the list, because completing the timers causes them to be removed
+        // from the list, which would result in a ConcurrentModificationException.
+        for (TransitionTimer timer : List.copyOf(transitionTimers)) {
+            timer.complete();
+        }
+    }
+
+    // package-private for testing
+    List<TransitionTimer> getTransitionTimers() {
+        return transitionTimers;
+    }
+
+    /**
+     * Contains descriptions of the animated transitions that are currently defined for
+     * properties of this {@code Node}.
+     * <p>
+     * All property transitions are implicit, which means they are started automatically by
+     * the CSS subsystem when a property value is changed. Explicit property changes, such as
+     * by calling {@link Property#setValue(Object)}, do not trigger an animated transition.
+     */
+    private class TransitionDefinitionCollection
+            extends ArrayList<TransitionDefinition>
+            implements StyleableProperty<TransitionDefinition[]> {
+        private StyleOrigin origin;
+
+        /**
+         * Returns the transition for the property referenced by the specified CSS metadata,
+         * or {@code null} if no transition was found.
+         *
+         * @param metadata the CSS metadata of the property
+         * @return the {@code TransitionDefinition} specified for the property referenced by the
+         *         CSS metadata, {@code null} otherwise
+         */
+        TransitionDefinition find(CssMetaData<? extends Styleable, ?> metadata) {
+            int size = size();
+            if (size == 0) {
+                return null;
+            }
+
+            // We look for a matching transition in reverse, since multiple transitions might be specified
+            // for the same property. In this case, the last transition takes precedence.
+            for (int i = size - 1; i >= 0; --i) {
+                TransitionDefinition transition = get(i);
+
+                boolean selected = "all".equals(transition.propertyName())
```

Comment:
Is "all" the same as `TransitionDefinitionCssMetaData.PROPERTY_ALL`, so the magic string can be avoided?

Reply:
I've replaced `"all"` with a named constant in all relevant places.